agent/structs/config_entry_routes.go

```diff
@@ -432,6 +436,18 @@ type URLRewrite struct {
 	Path string
 }
 
+type HTTPRouteRetry struct {
+	NumRetries            *uint32
+	RetryOn               *[]string
+	RetryOnStatusCodes    *[]uint32
+	RetryOnConnectFailure *bool
+}
+
+type HTTPRouteTimeout struct {
```

Comment:
minor: do we want postfix these names with `Filter` to make it clear they're filters on a route or do you think that's too verbose/not really necessary?

Reply:
The reason I didn't is because the filter types on the gateway spec didn't have the filter suffix. If its something we agree we should add I don't have strong feelings about it.

Reply:
I'm in favor of the `Filter` postfix, looking at the naming conventions in the gateway-api repo [over here](https://github.com/kubernetes-sigs/gateway-api/blob/62040436fa988cf5c9d13a992c3627810402fde9/apis/v1beta1/httproute_types.go#L557-L563)

Reply:
I'm in favor of the `Filter` suffix as well, and I think we can either drop `HTTPRoute` at the beginning in its entirety or just make it `HTTP` so either something like:

- `HTTPTimeoutFilter` or
- `TimeoutFilter` (if we want to re-use this on say GRPC protocols in the future)

Reply:
We landed on `HTTPRoute` prefix for the corresponding CRD over in consul-k8s so that the names are self-documenting. Things here in the consul repo don't have to match since we already have some filter naming conventions without the prefix.

Reply:
Changed it to TimeoutFilter and RetryFilter, and the CRD work will be named HTTPRouteTimeoutFilter and HTTPRouteRetryFilter 👍